### PR TITLE
User registration

### PR DESCRIPTION
**Pull request by johannaanden**
### PT Link
[PT Link](https://www.pivotaltracker.com/story/show/162193279)

### User Story
As a user,
In order to have be able to finalise an order
I would like to be able to register an account

### Actions
- Installed Devise
- Setup model users
- Added controller home
- Added routes for Devise views

### LoFi
Visit PT to see the LoFi
[PT Link](https://www.pivotaltracker.com/story/show/162193279)

### Gems added
- Devise

### What have we learned 
- How to implement Devise
- How to setup routs for Devise

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
```
Comment (oliverochman) [inline app/models/user.rb]: clean this file up

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,290 @@
+# frozen_string_literal: true
+
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
```
Comment (oliverochman) [inline config/initializers/devise.rb]: clean this one aswell

File: features/Step_definitions/user_registration_steps.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+Given("I click {string}") do |element|
+    click_on element
+end
+  
+Then("I am on landing page") do
+    expect(current_path).to eq root_path
+end
+
+Then("I am on users page") do
+    # binding.pry
```
Comment (oliverochman) [inline features/Step_definitions/user_registration_steps.rb]: remove commented out code

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (oliverochman) [inline spec/models/user_spec.rb]: You have no unit tests for the User model that you guys created in this PR
You need to add that

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (johannaanden) [reply app/models/user.rb]: done

File: config/initializers/devise.rb (hunk 2 above)
Comment (johannaanden) [reply config/initializers/devise.rb]: done

File: features/Step_definitions/user_registration_steps.rb (hunk 3 above)
Comment (johannaanden) [reply features/Step_definitions/user_registration_steps.rb]: done

File: spec/models/user_spec.rb (hunk 4 above)
Comment (johannaanden) [reply spec/models/user_spec.rb]: done

File: config/routes.rb [hunk 5]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
+
+  get 'home/index'
+  devise_for :users
+  root to: "home#index"
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
```
Comment (oliverochman) [inline config/routes.rb]: Remove line 3 and 5
add `root controller: :home, action: :index` instead

File: features/Step_definitions/user_registration_steps.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+Given("I am on user registration page") do
```
Comment (oliverochman) [inline features/Step_definitions/user_registration_steps.rb]: Rename this file to basic_steps and create another file named assertion_steps.
Put all step definitions where you expect stuff in the assertion_steps

File: features/Step_definitions/user_registration_steps.rb (hunk 3 above)
Comment (oliverochman) [reply features/Step_definitions/user_registration_steps.rb]: the commented out binding.pry is still here

File: app/views/home/index.html.haml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+%h1 Home#index
```
Comment (oliverochman) [inline app/views/home/index.html.haml]: remove the scaffolded code here

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/views/home/index.html.haml (hunk 7 above)
Comment (johannaanden) [reply app/views/home/index.html.haml]: done

File: features/Step_definitions/user_registration_steps.rb (hunk 6 above)
Comment (johannaanden) [reply features/Step_definitions/user_registration_steps.rb]: done

File: config/routes.rb (hunk 5 above)
Comment (johannaanden) [reply config/routes.rb]: done

File: features/Step_definitions/user_registration_steps.rb (hunk 3 above)
Comment (johannaanden) [reply features/Step_definitions/user_registration_steps.rb]: done

File: app/controllers/home_controller.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+class HomeController < ApplicationController
+  def index
```
Comment (oliverochman) [inline app/controllers/home_controller.rb]: is this controller still needed?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/home_controller.rb (hunk 8 above)
Comment (johannaanden) [reply app/controllers/home_controller.rb]: no, we have removed it now

File: features/user_registration.feature [hunk 9]
```diff
@@ -0,0 +1,23 @@
+Feature: User registration
+
+    As a user,
+    In order to have be able to finalise an order
+    I would like to be able to register an account
+
+Scenario: User registration [Happy path]
```
Comment (oliverochman) [inline features/user_registration.feature]: Indent the scenarios

**Review by oliverochman — CHANGES_REQUESTED** [review]
Last comment, fix this and we can merge it

File: features/user_registration.feature (hunk 9 above)
Comment (johannaanden) [reply features/user_registration.feature]: done

**Review by oliverochman — APPROVED** [review]
LGTM!

**Pull request merged**